### DISCUSSION TITLE
How to display correctly a logo using next/image

### DISCUSSION THREAD

**Discussion by geosigno** [vercel/next.js#24476] [Help]
I would need to implement a logo that will vary depending on the page. So the height and the width will also vary.

The thing that i know is the height that I want (100px), not the width. Is there any way to display the logo with the next/image component ? With a hardcoded 100px height, and an 'auto' width? At least to get the webP conversion?

I could get the logo dimension before displaying on the page, but to get it I need to download the image at some point, which will have the same result on the bandwidth than displaying a regular image tag with no optim at all.

**Comment by uxduck** [comment]
This is an existing issue, with 80 upvotes https://github.com/vercel/next.js/issues/18497

**Comment by bobarros** [comment]
As @ben-xD  already pointed out, you need to render something to know the final size. 
There's no auto before any rendering.

Sometimes, when it's really important to save some data, I try something like:

``` jsx

const ImageWrap= () => {
  const [loadImage, setLoad] = useState(false);
  const [finalWidth, setWidth] = useState(0);
  const [finalHeight, setHeight] = useState(0);

  useEffect(() => {    
    console.log("fire");
    const elHeight = document.getElementById("testImage").offsetHeight;
    const elWidth = document.getElementById("testImage").offsetWidth;
    
////any calc to set the height and width that I want.

    setHeight(elHeight);
    setWidth(elWidth);
    setLoad(true);

  },[]);

return (
<div>
        {!loadImage && <img id="testImage" src="/img/black.png" alt="test" />}
        {loadImage && <Image width={finalWidth} height={finalHeight} src="/img/image.jpg" alt="teste" />}
</div>
);
```

The black png is a very small image with something like some hundred bytes of the same [w x h] as my original file.

Don't forget to answer some questions before deciding if what you're doing it's a good idea:

1. Stretching the logo it's not an issue? 
2. Are you messing with the Cumulative Layout Shift (CLS)?
3. Are you trying too hard to save some Hbs or it's really important?
4. Are you making your code too complicated to read for no reason?
